### ISSUE TITLE
Question:  Why Didn't Time Advance?

### ISSUE THREAD

**Issue by billdenney** [metrumresearchgroup/mrgsolve#347]
It isn't clear to me why the first attempt here did not advance the time during simulation while the second worked as expected.  I think that I don't correctly understand the `tgrid` function and option.

```
simulate <-
  model %>%
  zero_re() %>%
  ev(ID=1, time=0, evid=1, cmt="CENT", amt=15*14, rate=15, doseratebase=15) %>%
  mrgsim(tgrid=tgrid(start=0, end=21, delta=0.25))
simulate

sim_data <-
  bind_rows(
    data_frame(ID=1,
               time=0,
               evid=1,
               cmt="CENT",
               rate=15,
               amt=rate*14,
               doseratebase=rate),
    data_frame(ID=1,
               time=seq(0, 21, by=0.25),
               evid=0,
               cmt="CENT"))

simulate <-
  model %>%
  zero_re() %>%
  data_set(sim_data) %>%
  mrgsim()
simulate
```

**Comment by kylebmetrum** [comment]
I'm getting identical results for both tries:

```r
> library(mrgsolve)
> library(dplyr)
> 
> model <- mrgsolve:::house()
> 
> simulate <-
+   model %>%
+   zero_re() %>%
+   ev(ID=1, time=0, evid=1, cmt="CENT", amt=15*14, rate=15, doseratebase=15) %>%
+   mrgsim(tgrid=tgrid(start=0, end=21, delta=0.25))
> 
> simulate
Model:  housemodel 
Dim:    86 x 7 
Time:   0 to 21 
ID:     1 
     ID time GUT   CENT  RESP     DV     CP
[1,]  1 0.00   0  0.000 50.00 0.0000 0.0000
[2,]  1 0.00   0  0.000 50.00 0.0000 0.0000
[3,]  1 0.25   0  3.727 49.80 0.1863 0.1863
[4,]  1 0.50   0  7.407 49.33 0.3704 0.3704
[5,]  1 0.75   0 11.042 48.72 0.5521 0.5521
[6,]  1 1.00   0 14.631 48.02 0.7316 0.7316
[7,]  1 1.25   0 18.176 47.30 0.9088 0.9088
[8,]  1 1.50   0 21.677 46.57 1.0838 1.0838
> 
> sim_data <-
+   bind_rows(
+     data_frame(ID=1,
+                time=0,
+                evid=1,
+                cmt="CENT",
+                rate=15,
+                amt=rate*14,
+                doseratebase=rate),
+     data_frame(ID=1,
+                time=seq(0, 21, by=0.25),
+                evid=0,
+                cmt="CENT"))
> 
> simulate2 <-
+   model %>%
+   zero_re() %>%
+   data_set(sim_data) %>%
+   mrgsim()
> 
> simulate2
Model:  housemodel 
Dim:    86 x 7 
Time:   0 to 21 
ID:     1 
     ID time GUT   CENT  RESP     DV     CP
[1,]  1 0.00   0  0.000 50.00 0.0000 0.0000
[2,]  1 0.00   0  0.000 50.00 0.0000 0.0000
[3,]  1 0.25   0  3.727 49.80 0.1863 0.1863
[4,]  1 0.50   0  7.407 49.33 0.3704 0.3704
[5,]  1 0.75   0 11.042 48.72 0.5521 0.5521
[6,]  1 1.00   0 14.631 48.02 0.7316 0.7316
[7,]  1 1.25   0 18.176 47.30 0.9088 0.9088
[8,]  1 1.50   0 21.677 46.57 1.0838 1.0838
> 
> 
> identical(simulate,simulate2)
[1] TRUE
```

**Comment by kylebmetrum** [comment]
The `tgrid` thing is just a abstraction of the time grid you get in the model; set a grid of times with `start/end/delta` and some additional times `add`.  

So these should be equivalent:

```r
tg <- tgrid(0,120,0.2, add = 33.3)
mrgsim(mod, tgrid = tg)
```

and 

```r
mrgsim(mod, end = 120, delta = 0.2, add = 33.3)
```

Just lets you set up different designs, captured as objects to use at the right time.

**Comment by kylebmetrum** [comment]
@billdenney I'm not sure what you were getting from the two scenarios.  Assuming it wasn't what I was getting.

**Comment by billdenney** [comment]
This was fixed with a related fix to #341 (a model-specific fix-- not related to the integrator).  It works now.
